src/cpu/aarch64/jit_primitive_conf.hpp

```diff
@@ -0,0 +1,723 @@
+    int kd, kh, kw;
+    int f_pad, t_pad, l_pad;
+    alg_kind_t alg;
+    bool is_training;
+    bool pad_w_is_null;
+    bool is_backward;
+    bool simple_alg;
+    bool is_c_padded;
+    data_type_t ind_dt;
+
+    int c_block, c_tail, nb_c;
+    int ur_bc, ur_bc_tail;
+    int ur_c, ur_c_tail;
+    int ur;
+    size_t tail[4];
+    bool safe_c_tail;
+    data_type_t src_dt;
+    data_type_t dst_dt;
+
+    int dt_size;
+    bool is_bf16;
+    jit_memory_tag_kind_t tag_kind;
+    bool is_plain() const {
+        return (tag_kind == jit_memory_tag_kind_t::ncsp
+                || tag_kind == jit_memory_tag_kind_t::nspc);
+    }
+
+    cpu_isa_t isa;
+    post_ops_t post_ops;
+    bool with_postops;
+    bool with_eltwise;
+    bool with_binary;
+};
+
+struct jit_pool_call_s {
+    const void *src;
+    const void *dst;
+    const void *indices;
+    const void *src_prf;
+    const void *dst_prf;
+    const void *indices_prf;
+    const void *post_ops_binary_rhs_arg_vec;
+    size_t c_elem_off;
+    size_t zero_ih;
+    size_t zero_id;
+    const void *zero_ptr;
+    size_t kd_padding;
+    size_t kh_padding;
+    size_t kh_padding_shift;
+    size_t kd_padding_shift;
+    size_t kw_padding;
+    const void *init_value;
+    float ker_area_h;
+    size_t ur_bc; // contains number of channel blocks to processing
+    size_t b_c; // contains number of channel blocks already processed
+};
+
+struct jit_resampling_conf_t {
+    unsigned ndims = 0;
+
+    unsigned id = 0, ih = 0, iw = 0;
+    unsigned od = 0, oh = 0, ow = 0;
+
+    unsigned stride_d = 0;
+    unsigned stride_h = 0;
+    unsigned stride_w = 0;
+    unsigned inner_stride = 0;
+
+    unsigned tail = 0;
+    unsigned simd_w = 0;
+
+    // The linear algorithm is an approximation of the point
+    // value based on the limit values. For one dimension,
+    // the approximation is based on the line, for two
+    // dimensions it will be a rectangle, and for three
+    // dimensions it will be a cuboid. Therefore,
+    // the possible variants for the number of corners are 2, 4, 8.
+    unsigned number_of_corners = 0;
+
+    bool is_data_size_bigger_than_L3 = false;
+    data_type_t data_type = data_type::undef;
+    size_t dt_size = 0;
+    size_t el_size_of_indices = 0;
+
+    jit_memory_tag_kind_t tag_kind = jit_memory_tag_kind_t::undef;
+    alg_kind_t alg = alg_kind::undef;
+
+    cpu_isa_t isa = isa_any;
+};
+
+struct jit_resampling_call_s {
+    size_t batch_of_sp_points_to_process = 0;
+
+    const void *src = nullptr;
+    const void *dst = nullptr;
+    const void *indices = nullptr;
+    const void *weights = nullptr;
+
+    size_t src_offset_top = 0;
+    size_t src_offset_bottom = 0;
+    size_t src_offset_front = 0;
+    size_t src_offset_back = 0;
+
+    float weight_top = 0.0f;
+    float weight_bottom = 0.0f;
+    float weight_front = 0.0f;
+    float weight_back = 0.0f;
+};
+
+} // namespace aarch64
+} // namespace cpu
+} // namespace impl
+} // namespace dnnl
+
+#endif
```

Comment:
For this entire file, I would suggest to only include the required fields.

Reply:
I second that!

Reply:
I'll delete unused code.
I will add it when I need it.